Support for LMDB flags

lquerel opened the pull request:
Added the support for the LMDB flags we discussed earlier.

Closes #15.

Laurent

Kerollmops commented:
Thank you for the contribution but isn’t a flush/sync function is necessary to force LMDB to persist data on disk if flags like `NoSync/NoMetaSync` is enabled?

Kerollmops commented on src/flags.rs:
```diff
@@ -0,0 +1,18 @@
+    MdbNoTls = 0x200000,
+    MdbNoLock = 0x400000,
+    MdbNoRdAhead = 0x800000,
+    MdbNoMemInit = 0x1000000,
```
I would prefer that you use the [`lmdb-rkv-sys` already extracted flags values](https://docs.rs/lmdb-rkv-sys/0.9.5/lmdb_sys/constant.MDB_FIXEDMAP.html) instead of redefining those by hand.

lquerel commented:
Thanks for your review, below a description of the modifications for this new PR:
- Added a force_sync method to the Env struct. 
- Reused the constant defined in lmdb_sys.
- Removed the dependency to enumflag2 because it was not compatible with the reference to lmdb_sys constants (enumflags2 only supports literals). However I kept a C-style enum for clarity and soundness.

Kerollmops commented:
Thank you for this pull request. I will squash your commits.

Sorry that I didn't provide a contributing guide, next time would it be possible to rebase your commits instead of merging master into your branch? 😊
